server/rest/server.go

```diff
@@ -25,12 +27,13 @@ func New(
 ) *http.Server {
 	return &http.Server{
 		Addr:              addr,
-		Handler:           buildServerHandler(readHandle, listHandle, createHandle, updateHandle),
+		Handler:           buildServerHandler(logger, readHandle, listHandle, createHandle, updateHandle),
```

Comment:
It's at this point where I'm beginning to think we might want a builder pattern for our handler here.  I wouldn't expect it for this PR, but I might end up doing this in the future.  Thoughts?